pkg/cloud/services/compute/instance.go

```diff
@@ -580,6 +582,20 @@ func (s *Service) InstanceDelete(machine *clusterv1.Machine, openStackMachine *i
 		return err
 	}
 
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceDelete, func() (bool, error) {
+		_, err = s.GetInstance(parsed.ID())
+		if err != nil {
+			if capoerrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
```

Comment:
It's fine for me but just to be sure. Do we want to stop polling here? Because returning the error will stop the PollImmediate func. Alternatively we could log the error and return nil so the PollImmedate tries again

Reply:
if it's error and not NOTFOUND error, this means something wrong happened...
I think we can stop polling otherwise we will always get timeout error

Reply:
Yup fine for me. I might want to adjust that later on a bit if we see other retryable errors in production, but absolutely fine for now